samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/SavePublishedSettingsListener.java

```diff
@@ -555,7 +555,13 @@ else if(!assessmentSettings.getIsValidFeedbackDate()){
 
 	// Check if the category has changed.
 	private boolean isCategoryChanged(PublishedAssessmentSettingsBean assessmentSettings, PublishedAssessmentFacade assessment) {
-		return !StringUtils.equals(assessmentSettings.getCategorySelected(), String.valueOf(assessment.getCategoryId()));
```

Comment:
Looks like the issue here is the return of String.valueOf(null)
NumberUtils probably simplifies this
```
Long selected = NumberUtils.toLong(assessmentSettings.getCategorySelected(), -1L);
return selected.equals(assessment.getCategoryId());
```

Reply:
assessment.getCategoryId() can be null or a String so can't use NumberUtils.toLong on it

Reply:
yeah it's on assessmentSettings.getCategorySelected() not assessment.getCategoryId() and assessment.getCategoryId() is a Long

Reply:
Initializing selected with a default and then using equals should be all that is needed as ```selected.equals(assessment.getCategoryId());``` will account for a null assessment.getCategoryId()

Reply:
if -1 equals 0 or null then it would trigger the re-creation of the gb item ....

Reply:
OK I see! so then:
```
Long selected = NumberUtils.toLong(assessmentSettings.getCategorySelected(), -1L);
return selected.equals(assessment.getCategoryId() != null ? assessment.getCategoryId() : -1L);
```

Reply:
I guess that doesn't cover 0!

Reply:
So lets simplify your if statement with:
```return (oldCatId > 0 || newCatId > 0) && oldCatId.compareTo(newCatId) != 0;```

Reply:
agree